vsintegration/src/FSharp.Editor/FindReferencesService.fs

```diff
@@ -0,0 +1,121 @@
+                    | Some symbolUse ->
+                        let! declaration = checkFileResults.GetDeclarationLocationAlternate (fcsTextLineNumber, islandColumn, textLine.ToString(), qualifiers, false)
+                        let declarationRange = 
+                            match declaration with
+                            | FSharpFindDeclResult.DeclFound range -> Some range
+                            | _ -> None
+                        
+                        let! declarationSpan =
+                            match declarationRange with
+                            | Some range -> rangeToDocumentSpan range
+                            | None -> async.Return None
+                
+                        let definitionItem =
+                            match declarationSpan with 
+                            | Some span ->
+                                DefinitionItem.Create(
+                                    ImmutableArray<string>.Empty, 
+                                    [TaggedText(TextTags.Text, symbolUse.Symbol.FullName)].ToImmutableArray(), 
+                                    span)
+                            | None -> 
+                                DefinitionItem.CreateNonNavigableItem(
+                                    ImmutableArray<string>.Empty, 
+                                    [TaggedText(TextTags.Text, symbolUse.Symbol.FullName)].ToImmutableArray())    
+                
+                        do! context.OnDefinitionFoundAsync(definitionItem) |> Async.AwaitTask
+                        
+                        let! symbolUses =
+                            async {
+                                if symbolUse.Symbol.IsPrivateToFile then
+                                    return! checkFileResults.GetUsesOfSymbolInFile(symbolUse.Symbol)
+                                else
+                                    let! projectCheckResults = checker.ParseAndCheckProject(options)
+                                    return! projectCheckResults.GetUsesOfSymbol(symbolUse.Symbol)
+                            }
+
+                        for symbolUse in symbolUses do
+                            match declarationRange with
+                            | Some declRange when declRange = symbolUse.RangeAlternate -> ()
+                            | _ ->
+                                let! referenceDocSpan = rangeToDocumentSpan symbolUse.RangeAlternate
+                                match referenceDocSpan with
+                                | Some span -> 
+                                    let referenceItem = SourceReferenceItem(definitionItem, span)
+                                    do! context.OnReferenceFoundAsync(referenceItem) |> Async.AwaitTask
+                                | None -> ()
+                    | None -> ()
+                | None -> ()
+            | None -> ()
+            
+            do! context.OnCompletedAsync() |> Async.AwaitTask
```

Comment:
When you move to the new interface, you will no longer have to call OnCompleted.  Your responsibility will just be reporting Definitions, Reference and Progress.